src/main/scala/io/citrine/lolo/PredictionResult.scala

```diff
@@ -44,6 +46,112 @@ trait PredictionResult[+T] {
   def getGradient(): Option[Seq[Vector[Double]]] = None
 }
 
+/**
+ * Additional regression-specific interface
+ */
```

Comment:
Suggest to add: "This interface is experimental and SHOULD BE REVIEWED before being merged into `master`. In particular, an explanation of how the different methods relate to each other, how predictive uncertainty is decomposed, and what the assumptions (*) are should be added, as these are currently not entirely clear.
(*) For example, does the interface assume that the predictions are the mean of a predictive distribution (as opposed to, for example, the median, or the value with highest probability)? Does it assume the predictive distribution to be normal? Such assumptions are fine, but should be explicitly stated."